src/Validator/RequestValidator.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace ZfrShopify\Validator;
+
+use Psr\Http\Message\ServerRequestInterface;
+use ZfrShopify\Exception;
+
+/**
+ * Validate an incoming request coming from Shopify
+ *
+ * @author Michaël Gallego
+ */
+class RequestValidator
+{
+    /**
+     * @link   https://docs.shopify.com/api/authentication/oauth#verification
+     * @param  ServerRequestInterface $request
+     * @param  string                 $sharedSecret
+     * @throws Exception\InvalidRequestException
+     */
+    public function validateRequest(ServerRequestInterface $request, string $sharedSecret)
+    {
+        // First step: extract the query params
+        $queryParams = $request->getQueryParams();
+
+        $this->validateShopHostname($queryParams);
+        $this->validateHmac($queryParams, $sharedSecret);
+    }
+
+    /**
+     * According to Shopify, a shop hostname must ends by "myshopify.com", and must only contains
+     * letters, numbers, dots and hyphens
+     *
+     * @param  array $queryParams
+     * @return void
+     * @throws Exception\InvalidRequestException
+     */
+    private function validateShopHostname(array $queryParams)
```

Comment:
Benefit is that we can use this value object to hold shop domain everywhere across the apps/microservices